commit e7cd05f2ae970826302285ae922732f89a5ee1c7
Author: Maxwell Pray <synthead@gmail.com>
Date:   Wed Oct 29 17:15:45 2014 -0700

    Add support for ten-key with respect to Num Lock state.

diff --git a/interface.c b/interface.c
index 163a45a..56470a2 100644
--- a/interface.c
+++ b/interface.c
@@ -1426,7 +1426,8 @@ static bool handle_key(SDLKey key, SDLMod mod, Uint16 unicode_key)
             selector_toggle(sel);
         return true;
 
-    } else if (mod & KMOD_CTRL && (key == SDLK_EQUALS || key == SDLK_PLUS)) {
+    } else if (mod & KMOD_CTRL &&
+               (key == SDLK_EQUALS || key == SDLK_PLUS || key == SDLK_KP_PLUS)) {
         meter_scale--;
 
         if (meter_scale < 0)
@@ -1434,7 +1435,7 @@ static bool handle_key(SDLKey key, SDLMod mod, Uint16 unicode_key)
 
         fprintf(stderr, "Meter scale decreased to %d\n", meter_scale);
 
-    } else if (mod & KMOD_CTRL && key == SDLK_MINUS) {
+    } else if (mod & KMOD_CTRL && (key == SDLK_MINUS || key == SDLK_KP_MINUS)) {
         meter_scale++;
 
         if (meter_scale > MAX_METER_SCALE)
@@ -1446,35 +1447,35 @@ static bool handle_key(SDLKey key, SDLMod mod, Uint16 unicode_key)
         selector_search_expand(sel);
         return true;
 
-    } else if (key == SDLK_HOME) {
+    } else if (key == SDLK_HOME || (key == SDLK_KP7 && !(mod & KMOD_NUM))) {
         selector_top(sel);
         return true;
 
-    } else if (key == SDLK_END) {
+    } else if (key == SDLK_END || (key == SDLK_KP1 && !(mod & KMOD_NUM))) {
         selector_bottom(sel);
         return true;
 
-    } else if (key == SDLK_UP) {
+    } else if (key == SDLK_UP || (key == SDLK_KP8 && !(mod & KMOD_NUM))) {
         selector_up(sel);
         return true;
 
-    } else if (key == SDLK_DOWN) {
+    } else if (key == SDLK_DOWN || (key == SDLK_KP2 && !(mod & KMOD_NUM))) {
         selector_down(sel);
         return true;
 
-    } else if (key == SDLK_PAGEUP) {
+    } else if (key == SDLK_PAGEUP || (key == SDLK_KP9 && !(mod & KMOD_NUM))) {
         selector_page_up(sel);
         return true;
 
-    } else if (key == SDLK_PAGEDOWN) {
+    } else if (key == SDLK_PAGEDOWN || (key == SDLK_KP3 && !(mod & KMOD_NUM))) {
         selector_page_down(sel);
         return true;
 
-    } else if (key == SDLK_LEFT) {
+    } else if (key == SDLK_LEFT || (key == SDLK_KP4 && !(mod & KMOD_NUM))) {
         selector_prev(sel);
         return true;
 
-    } else if (key == SDLK_RIGHT) {
+    } else if (key == SDLK_RIGHT || (key == SDLK_KP6 && !(mod & KMOD_NUM))) {
         selector_next(sel);
         return true;
 